backend/db_funcs/cipherGameFuncs.py

```diff
@@ -9,33 +9,77 @@
-                 updated_cipher_game.deadline_event, updated_cipher_game.capacity, updated_cipher_game.teammax, updated_cipher_game.password, updated_cipher_game.autoapprove,
-                 cipher_game_id))
+def updateCipherGame(cipher_game_id: int, updated_cipher_game: Cipher_game):
+    try:
+        with DB_conn.getConn(connection):
+            with DB_conn.getCursor(connection) as cur:
+                cur.execute(
+                    "UPDATE cipher_game SET cipher_id = %s, name = %s, description = %s, visible_from = %s, deadline_signup = %s, deadline_event = %s, capacity = %s, teammax = %s, password = %s, autoapprove = %s WHERE cipher_game_id = %s;",
+                    (updated_cipher_game.cipher_id, updated_cipher_game.name, updated_cipher_game.description, updated_cipher_game.visible_from, updated_cipher_game.deadline_signup,
+                     updated_cipher_game.deadline_event, updated_cipher_game.capacity, updated_cipher_game.teammax, updated_cipher_game.password, updated_cipher_game.autoapprove,
+                     cipher_game_id))
+    except:
+        return {"result": "error"}
+    return {"result": "updated"}
 
 
-def delete_cipher_game(cipher_game_id: int):
-    with DB_conn.getConn(connection):
-        with DB_conn.getCursor(connection) as cur:
-            success = cur.execute("DELETE FROM cipher_game WHERE cipher_game_id = %s;", (cipher_game_id,))
+def deleteCipherGame(cipher_game_id: int):
+    try:
+        with DB_conn.getConn(connection):
+            with DB_conn.getCursor(connection) as cur:
+                success = cur.execute("DELETE FROM cipher_game WHERE cipher_game_id = %s;", (cipher_game_id,))
+    except:
+        return {"result": "error"}
+    return {"result": success}
+
+
+def getCipherGames():
+    try:
+        with DB_conn.getConn(connection):
+            with DB_conn.getCursor(connection) as cur:
+                cur.execute("SELECT * FROM cipher_game;")
+                result = cur.fetchall()
+    except:
+        return {"result": "error"}
+    return result
+
+
+def get_cipher_game(cipher_game_id: int):
+    if not is_visible(cipher_game_id):
+        return None
+    try:
+        with DB_conn.getConn(connection):
+            with DB_conn.getCursor(connection) as cur:
+                cur.execute("SELECT * FROM cipher_game WHERE cipher_game_id = %s;", (cipher_game_id,))
+                result = cur.fetchall()
+    except:
+        return None
+    return result
+
+
+def is_visible(cipher_game_id: int):
+    timestamp = datetime.now()
+    try:
+        with DB_conn.getConn(connection):
+            with DB_conn.getCursor(connection) as cur:
+                cur.execute("SELECT * FROM cipher_game WHERE visible_from >=  %s AND cipher_game_id = %s;", (timestamp,cipher_game_id))
+                result = cur.fetchall()
+    except:
+        return False
+    return bool(result)
```

Comment:
Ještě bychom asi chtěli přidat kontrolu, jestli uživatel není root, nebo admin dané hry, protože potom ji vidí.

Takže by možná dávalo smysl přesunout tuhle kontrolu výš. Postarám se o to.